src/App.jsx

```diff
@@ -13,34 +16,106 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORTED_BY_ALFABET = 'Sort alphabetically';
+const SORT_FIELD_LENGTH = 'length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, { sortField, reset, reverse }) {
+  let preperedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORTED_BY_ALFABET:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preperedGoods.reverse();
+  }
+
+  if (reset) {
+    preperedGoods = goods;
+  }
```

Comment:
minor thing, but it makes sense to perform this check in the start of the function (to avoid unnecessary manipulations)

Reply:
And more - I don't see you ever actually pass `reset` field to `getPreparedGoods`

Reply:
Hello Alex, thanks for check. This is my inattention. Delete reset.